<?xml version="1.0"?>

<launch>

  <!-- For sending goal -->
  <arg name="goal_topic"                default="/move_base_simple/goal"/>
  <arg name="goal_frame"                default="map"/>
  <arg name="send_goal"                 default="false"/>
  <!-- For object detection -->
  <arg name="segmented_image_topic"     default="/color_seg"/>
  <!-- Percentage [0%, 100%] of pixels passing segmentation for detection to pass -->
  <!-- Very small values (x << 5 ) recommended -->
  <arg name="detection_threshold"       default="1"/>


  <node name="detect_stop_sign" pkg="flight_camp" type="detect_stop_sign.py" output="screen">
    <!-- For sending goal -->
    <param name="goal_topic"                            value="$(arg goal_topic)"/>
    <param name="goal_frame"                            value="$(arg goal_frame)"/>
    <param name="send_goal"             type="bool"     value="$(arg send_goal)"/>
    <!-- For object detection -->
    <param name="segmented_image_topic"                 value="$(arg segmented_image_topic)"/>
    <param name="detection_threshold"   type="double"   value="$(arg detection_threshold)"/>

  </node>

</launch>

